api/src/main/java/com/community/api/services/QualificationDetailsService.java

```diff
@@ -11,166 +14,218 @@
-            throw new EntityAlreadyExistsException("Qualification with id " + qualificationDetails.getQualification_id() + " already exists");
+        if (roleName.equals(Constant.SERVICE_PROVIDER)) {
+            ServiceProviderEntity serviceProviderEntity = findServiceProviderById(userId);
+            checkIfQualificationAlreadyExists(userId, qualificationDetails.getQualification_id(), roleName);
+            List<DocumentType> qualifications = qualificationService.getAllQualifications();
+            Integer qualificationToAdd = findQualificationId(qualificationDetails.getQualification_id(), qualifications);
+            qualificationDetails.setQualification_id(qualificationToAdd);
+            qualificationDetails.setService_provider(serviceProviderEntity);
+            serviceProviderEntity.getQualificationDetailsList().add(qualificationDetails);
+            entityManager.persist(qualificationDetails);
+            return qualificationDetails;
         }
+        CustomCustomer customCustomer = findCustomCustomerById(userId);
+        checkIfQualificationAlreadyExists(userId, qualificationDetails.getQualification_id(), roleName);
         List<DocumentType> qualifications = qualificationService.getAllQualifications();
-        Integer examinationToAdd=null;
-
-        for(DocumentType qualification : qualifications)
-        {
-            if(qualification.getDocument_type_id()== qualificationDetails.getQualification_id()) {
-                examinationToAdd= qualification.getDocument_type_id();
-                break;
-            }
-        }
-        if (examinationToAdd == null) {
-            throw new ExaminationDoesNotExistsException("Qualification with id " + qualificationDetails.getQualification_id() + " does not exist");
-        }
-        qualificationDetails.setQualification_id(examinationToAdd);
+        Integer qualificationToAdd = null;
+        qualificationToAdd = findQualificationId(qualificationDetails.getQualification_id(), qualifications);
+        qualificationDetails.setQualification_id(qualificationToAdd);
         qualificationDetails.setCustom_customer(customCustomer);
         customCustomer.getQualificationDetailsList().add(qualificationDetails);
         entityManager.persist(qualificationDetails);
         return qualificationDetails;
     }
 
-    public List<QualificationDetails> getQualificationDetailsByCustomerId(Long customCustomerId) throws  CustomerDoesNotExistsException,RuntimeException {
-        CustomCustomer customCustomer= entityManager.find(CustomCustomer.class,customCustomerId);
-        if(customCustomer==null)
-        {
-            throw new CustomerDoesNotExistsException("Customer does not exist with id "+ customCustomerId);
+    @Transactional
+    public List<Map<String, Object>> getQualificationDetailsByCustomerId(Long userId, String roleName) throws CustomerDoesNotExistsException, RuntimeException {
+        List<QualificationDetails> qualificationDetails;
```

Comment:
here if someone enters string
give NumberFormatException exception

use this
Long id = Long.valueOf(customerId);

catch (NumberFormatException e) {
    return ResponseService.generateErrorResponse("Invalid customerId: expected a Long", HttpStatus.BAD_REQUEST);
}